magefile.go

```diff
@@ -1571,6 +1573,41 @@ func (Integration) Single(ctx context.Context, testName string) error {
 	return integRunner(ctx, false, testName)
 }
 
+// UpdateVersions runs an update on the `.agent-versions.json` fetching
+// the latest version list from the artifact API.
+func (Integration) UpdateVersions(ctx context.Context) error {
+	// test 2 current 8.x version, 1 previous 7.x version and 1 recent snapshot
+	reqs := upgradetest.VersionRequirements{
+		UpgradeToVersion: bversion.Agent,
+		CurrentMajors:    2,
+		PreviousMinors:   1,
+		PreviousMajors:   1,
+		RecentSnapshots:  1,
+	}
+
+	aac := tools.NewArtifactAPIClient()
+	versions, err := upgradetest.FetchUpgradableVersions(ctx, aac, reqs)
+	if err != nil {
+		return fmt.Errorf("failed to fetch upgradable versions: %w", err)
+	}
+	versionFileData := upgradetest.AgentVersions{
+		TestVersions: versions,
+	}
+	file, err := os.OpenFile(upgradetest.AgentVersionsFilename, os.O_CREATE|os.O_TRUNC|os.O_WRONLY, 0644)
```

Comment:
[Nit]
You can use [`os.WriteFile`](https://pkg.go.dev/os#WriteFile)

Reply:
I prefer to use streaming, instead of buffering. In this particular case it does not matter match, I agree, but I prefer to keep it this way.